ethers-core/src/types/transaction/eip1559.rs

```diff
@@ -42,7 +42,12 @@ pub struct Eip1559TransactionRequest {
 
     /// The compiled code of a contract OR the first 4 bytes of the hash of the
     /// invoked method signature and encoded parameters. For details see Ethereum Contract ABI
-    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_input_or_data",
+        serialize_with = "serialize_input_or_data",
+        flatten
```

Comment:
I think this is just `alias = "input"`?

Reply:
That's what I tried to do as well, however it fails on the example where both fields are present:
```
---- types::transaction::eip1559::tests::test_tx_with_input_or_data stdout ----
thread 'types::transaction::eip1559::tests::test_tx_with_input_or_data' panicked at 'failed to parse request { "gas": "0x186a0",
                    "maxFeePerGas": "0x77359400",
                    "maxPriorityFeePerGas": "0x77359400",
                    "nonce": "0x2",
                    "to": "0x96216849c49358B10257cb55b28eA603c874b05E",
                    "value": "0x5af3107a4000",
                    "chainId": "0x539",
                    "accessList": [],
                     "data":"0x02", "input":"0x01" 
                }: duplicate field `data` at line 9 column 43', ethers-core/src/types/transaction/eip1559.rs:340:37
```

I'm not sure how important that is, whether there are actual clients that are so defensive that they send both fields, not knowing which one the server might expect; I only got it as a hint from someone who debugged this problem that certain implementation have a _preference_ for `input`, which presumes that there is a choice to made.

Reply:
sigh

Reply:
Just saw in https://github.com/web3/web3.js/issues/1340 that Parity rejects the request if both `input` and `data` is present, so maybe it's okay to require one or the other with a simple `alias`.

Reply:
Duplication used to be an issue with web3 but got fixed apparently: https://github.com/web3/web3.js/issues/6301

Reply:
Switched to `alias`, and added it to the legacy transaction type as well, just in case.